moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -488,3 +504,26 @@ void planning_scene_monitor::CurrentStateMonitor::tfCallback()
     state_update_condition_.notify_all();
   }
 }
+
+// Copied from https://github.com/ros2/geometry2/blob/ros2/tf2_ros/src/transform_listener.cpp
+void planning_scene_monitor::CurrentStateMonitor::transfromCallback(const tf2_msgs::msg::TFMessage::ConstSharedPtr msg,
+                                                                    const bool is_static)
+{
+  for (const auto& transform : msg->transforms)
+  {
+    try
+    {
+      tf_buffer_->setTransform(transform,
+                               is_static ? static_transfrom_subscriber_->get_topic_name() :
+                                           transform_subscriber_->get_topic_name(),
+                               is_static);
+    }
+    catch (tf2::TransformException& ex)
+    {
+      std::string temp = ex.what();
+      RCLCPP_ERROR(LOGGER, "Failure to set recieved transform from %s to %s with error: %s\n",
+                   transform.child_frame_id.c_str(), transform.header.frame_id.c_str(), temp.c_str());
+    }
+  }
+  updateMultiDofJoints();
```

Comment:
isn't this redundant in case of a transform exception?

Reply:
What if the exception happened after adding few transforms to the buffer .?